engines/sci/resource/decompressor.cpp

```diff
@@ -176,172 +165,80 @@ int DecompressorLZW::unpack(Common::ReadStream *src, byte *dest, uint32 nPacked,
 	return 0;
 }
 
-int DecompressorLZW::unpackLZW(Common::ReadStream *src, byte *dest, uint32 nPacked,
-								uint32 nUnpacked) {
+// Decompresses SCI0 LZW and SCI01/1 LZW, depending on _compression value.
+//
+// SCI0:    LSB-first.
+// SCI01/1: MSB-first, code size is increased one code earlier than necessary.
+//          This is known as an "early change" bug in LZW implementations.
```

Comment:
I'm a bit confused here. The "early change" bug was introduced in early implementations of the LZW algorithm, but here it's introduced in later versions. Was this introduced by accident, perhaps?

Reply:
Yes, it's very confusing that their second LZW implementation introduced this bug! I don't have any insight beyond that; it's a common mistake to make in LZW but it doesn't really matter as long as both the encoder and decoder behave the same.

My guess is that Sierra didn't know they  introduced this bug in their second implementation. They already swapped the bit order, so it's not like any of the old stuff and new stuff would have been compatible. But it is funny that if it weren't for this, the only difference would be "flip the bits".

Reply:
Odd indeed. Thanks for the explanation :)